### PR TITLE
Add button to delete listings

### PR DESCRIPTION
**Pull request by woeltjen**
## Description
This provides hosts with the ability to delete listings they have created

## How to Test
1. Create a new listing
2. Go to /host/listings
3. Locate the created listing
4. Click "Delete"
5. Confirm that you wish to delete via the popup
6. Expect the deleted listing to disappear

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
If we want more buttons, we're running out of room on that card...

## Learnings
Our `Button` gives a `min-width` of `120px` to its contained `button`, if you're ever having trouble sizing buttons...

### CODE REVIEW

**Comment by jerminatorhits** [comment]
successfully works as an admin user!

For regular user, I clicked `delete` on my listing and it failed silently. The frontend console throws the `not authorized` error due to the backend admin guard. Small backend change needs to allow `listing.hostId` to match user `id` as well.

**Comment by woeltjen** [comment]
@jerminatorhits Thanks for catching that! I'll submit a backend PR as well

File: src/components/routes/Host/HostListingCard/HostListingCard.container.ts [hunk 1]
```diff
@@ -30,10 +30,12 @@ const HostListingCardContainer = styled.article`
...
       justify-content: space-between;
-      width: 504px;
-      button,
-      a {
-        width: 127px;
+      width: 555px;
+      a,
+      button {
+        min-width: 105px;
+        padding: 0px;
```
[the remark this answers is not in the document]
Comment (woeltjen) [reply src/components/routes/Host/HostListingCard/HostListingCard.container.ts]: Thanks for pointing that out. It looks like this `padding` wasn't relevant to the way these were displaying, so I've removed it.
